adagrams/game.py

```diff
@@ -1,11 +1,128 @@
+import random
+
 def draw_letters():
-    pass
+    LETTER_POOL = {
+        "A": 9,
+        "B": 2,
+        "C": 2,
+        "D": 4,
+        "E": 12,
+        "F": 2,
+        "G": 3,
+        "H": 2,
+        "I": 9,
+        "J": 1,
+        "K": 1,
+        "L": 4,
+        "M": 2,
+        "N": 6,
+        "O": 8,
+        "P": 2,
+        "Q": 1,
+        "R": 6,
+        "S": 4,
+        "T": 6,
+        "U": 4,
+        "V": 2,
+        "W": 2,
+        "X": 1,
+        "Y": 2,
+        "Z": 1
+    }
+
+    hand_letters = []
+    letters = list(LETTER_POOL.keys())
+    
+    while len(hand_letters) < 10:
+        letter = random.choice(letters)
+        if LETTER_POOL[letter] > 0:
+            LETTER_POOL[letter] -= 1
+            hand_letters.append(letter)
+    return hand_letters
 
 def uses_available_letters(word, letter_bank):
-    pass
+    copy_letter_bank = letter_bank[:]
+
+    for letter in word.upper():
+        if letter in copy_letter_bank:
+            copy_letter_bank.remove(letter)
+        else:
+            return False
+    return True
 
 def score_word(word):
-    pass
+    SCORE_CHART = {
+        1: ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
+        2: ["D", "G"],
+        3: ["B", "C", "M", "P"],
+        4: ["F", "H", "V", "W", "Y"],
+        5: ["K"],
+        8: ["J", "X"],
+        10: ["Q", "Z"]
+    }
+
+    score = 0
+    score_chart_letters = list(SCORE_CHART.values())
+    word_letters = list(word.upper())
+    index_to_points = []
+
+    for i in range(len(score_chart_letters)):
```

Comment:
It's not a rule that works in all cases because sometimes you just need to do it, but I'd recommend that if you find you're needing 3 levels of looping, give some thought to alternative approaches, perhaps considering modifying the original data structure! For example, the structure I showed above, you could calculate the score with 2 loops:
```python
score = 0
for letter in word.upper():
  for str in LETTER_VALUES:
    if letter in str:
      score += LETTER_VALUES[str]
```
Or if you use the dictionary letter -> point value mapping, you can do it in 1 loop!
```python
   score = 0
    for letter in word:
        score += LETTER_VALUES[letter.upper()]
```